src/ray/common/scheduling/scheduling_ids.cc

```diff
@@ -0,0 +1,50 @@
+#include "scheduling_ids.h"
+
+int64_t ScheduleIds::getIdByInt(string sid) {
+  if (string_to_int.count(sid) == 0) return 0;
+  return string_to_int[sid];
+};
+
+int64_t ScheduleIds::insertIdByString(string sid) {
+  auto sit = string_to_int.find(sid);
+  if (sit == string_to_int.end()) {
+    int64_t id = hasher(sid);
+    for (int i = 0; true; i++) {
+      auto it = int_to_string.find(id);
+      if (it == int_to_string.end()) {
+        /// No hash collision, so associated sid with id.
+        string_to_int.insert(make_pair(sid, id));
```

Comment:
can do `string_to_int.emplace(sid, id);` to avoid copy of the pair (same for next line)